lua/nvim-lsp-installer/servers/omnisharp/init.lua

```diff
@@ -17,38 +18,40 @@ return function(name, root_dir)
             context.use_github_release_file(
                 "OmniSharp/omnisharp-roslyn",
                 coalesce(
-                    when(platform.is_mac, "omnisharp-osx.zip"),
+                    when(
+                        platform.is_mac,
+                        coalesce(
+                            when(platform.arch == "x64", "omnisharp-osx-x64-net6.0.zip"),
+                            when(platform.arch == "arm64", "omnisharp-osx-arm64-net6.0.zip")
+                        )
+                    ),
                     when(
                         platform.is_linux,
                         coalesce(
-                            when(platform.arch == "x64", "omnisharp-linux-x64.zip"),
-                            when(platform.arch == "arm64", "omnisharp-linux-arm64.zip")
+                            when(platform.arch == "x64", "omnisharp-linux-x64-net6.0.zip"),
+                            when(platform.arch == "arm64", "omnisharp-linux-arm64-net6.0.zip")
                         )
                     ),
                     when(
                         platform.is_win,
                         coalesce(
-                            when(platform.arch == "x64", "omnisharp-win-x64.zip"),
-                            when(platform.arch == "arm64", "omnisharp-win-arm64.zip")
+                            when(platform.arch == "x64", "omnisharp-win-x64-net6.0.zip"),
+                            when(platform.arch == "arm64", "omnisharp-win-arm64-net6.0.zip")
                         )
                     )
                 )
             ),
             context.capture(function(ctx)
                 return std.unzip_remote(ctx.github_release_file, "omnisharp")
             end),
-            std.chmod("+x", { "omnisharp/run" }),
             context.receipt(function(receipt, ctx)
                 receipt:with_primary_source(receipt.github_release_file(ctx))
             end),
         },
         default_options = {
             cmd = {
-                platform.is_win and path.concat { root_dir, "omnisharp", "OmniSharp.exe" } or path.concat {
-                    root_dir,
-                    "omnisharp",
-                    "run",
-                },
+                "dotnet",
+                path.concat { root_dir, "OmniSharp.dll" },
```

Comment:
I believe the following fixes the issues for the Mac archive (we'll have to make sure the structure of each .zip archive is the same though):

```suggestion
                path.concat { root_dir, "omnisharp", "OmniSharp.dll" },
```

Reply:
Ah so all the zip files are luckily of the same structure. The reason `omnisharp` is needed here is because the [unzip is currently instructed to unzip to the `omnisharp` directory](https://github.com/williamboman/nvim-lsp-installer/blob/455de5e7a608b830b1756403b73322d8844b50f6/lua/nvim-lsp-installer/servers/omnisharp/init.lua#L45)

Reply:
Thanks for that! I didn't notice while testing that I had the path hardcoded in my config. That's why it didn't pick up the default from here.